medication statements/README.md

```diff
@@ -0,0 +1,363 @@
+| effectivePeriod.end                              | `ehrSupplyDiscontinue / availabilityTime [@value]` <sup>8</sup>                                                                                                            |
+| effectivePeriod.start                            | `ehrSupplyAuthorise / effectiveTime / center [@value]`, `ehrSupplyAuthorise / effectiveTime / low [@value]`, `ehrSupplyAuthorise / availabilityTime [@value]` <sup>9</sup> |
+| dateAsserted                                     | `ehrComposition / author / time [@value]`, `ehrExtract / availabilityTime [@value]`                                                                                        |
+| subject                                          | A reference to the [Patient](../patient/README.md) from `ehrExtract / recordTarget / patient`                                                                              |
+| context                                          | A reference to the [Encounter](../encounters/README.md) which this Medication Statement refers to.                                                                         |
+
+1. Taken is included as both a mapped field and field not in use.  The reason for this being that the item is mandatory in the base FHIR profile but GP systems do not record this detail. For this reason a default value of 'unk' is picked but should **not** be used</br>
+2. If there is no `pertinentInformation` which has a `pertinentInformationDosage / text`, then a default value of `"No Information available"` is used
+3. This uses the most recent `availabilityTime` of an `ehrSupplyPrescribe` within any medication statement in the `ehrExtract` where there is an `inFulfilmentOf` which has a `priorMedication / id [@root]` matching the id of the current `MedicationStatement`<br/> If these are no matches then this extension is not added
+4. This uses the values of `[@code]`, `[@displayName]` and `originalText` from `MedicationStatement / consumable / manufacturedProduct / manufacturedMaterial / code[0]` to create a unique key with a UUID value.  If this key has been used before then a `Medication` reference with the previously created UUID is used, otherwise a new UUID is generated. 
+5. If there are no `consumable` with a `manufacturedProduct` with a `manufacturedMaterial` then this value is not set
+6. The `ehrExtract` is parsed . If this has `ehrSupplyDiscontinue / availabilityTime / value` then value is set to `"STOPPED"` or `"COMPLETED"`
+7. If there is no matching `ehrSupplyDiscontinue` (see <sup>6</sup>) then if `ehrSupplyAuthorise / statusCode / code` = `"COMPLETE"` then `"COMPLETED"` is used, else `"ACTIVE"` is used
+8. When `ehrSupplyDiscontinue / availabilityTime [@value]` does not exist and `status` = `"ACTIVE"` then this is set to the same value as `effectivePeriod.start`
```

Comment:
Seems to fallback to `ehrSupplyAuthorise / effectiveTime / high \[@value]` and then falls back further to `medicationStatement / effectiveTime \[@value]`

Reply:
Oh wait, I can see the Active behaviour now.  Looks like that's in addition to the fallbacks above.

Reply:
Yeah I agree, I've run through a number of scenarios and I can't see any circumstances where this logic will come into play, have removed this for clarity and added the fallback scenarios